enterprise_gateway/services/processproxies/crd.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+"""Code related to managing kernels running based on k8s custom resource."""
+
+from .k8s import KubernetesProcessProxy
+from kubernetes import client
+
+
+class CustomResourceProxy(KubernetesProcessProxy):
+    group = version = plural = None
+    custom_resource_template_name = None
+    kernel_resource_name = None
+
+    def __init__(self, kernel_manager, proxy_config):
+        super(CustomResourceProxy, self).__init__(kernel_manager, proxy_config)
+
+    async def launch_process(self, kernel_cmd, **kwargs):
+        kwargs['env']['KERNEL_RESOURCE_NAME'] = self.kernel_resource_name = self._determine_kernel_pod_name(**kwargs)
+        kwargs['env']['KERNEL_CRD_GROUP'] = self.group
+        kwargs['env']['KERNEL_CRD_VERSION'] = self.version
+        kwargs['env']['KERNEL_CRD_PLURAL'] = self.plural
+
+        await super(CustomResourceProxy, self).launch_process(kernel_cmd, **kwargs)
+        return self
+
+    def kill(self):
+        result = None
+
+        if self.kernel_resource_name:
+            result = self.terminate_custom_resource()
+
+        return result
```

Comment:
Does this require a call to its superclass in order to terminate the container resources - like namespace, or does `terminate_custom_resource()` handle all of that?

How is the namespace lifecycle managed with custom resources?

Reply:
sorry for forgot it, namespace should be removed when such action required